Bubble sort algorithm pull request

temesghentekeste opened the pull request:
In the initial configuration of the project, we created a feature branch called bubble-sort-dev. In that branch, we set up GitHub workflows and added linters.yml and test.yml inside .github/workflows folder.

After adding .rubocop.yml in the root directory of the project, we developed the logic of the sorting algorithm inside bubble-sort.rb file; we also created main.rb file to start the sorting functionality passing an unsorted array. We used two files to achieve separation of concerns between the sorting logic and the implementation of the logic.

We kindly request a pull request for our branch and thanks in advance.

suretrust commented on bubble_sort.rb:
```diff
@@ -0,0 +1,21 @@
+# in the inner loop two adjacent elemens are compared
+# if element at lower position is greater than the next one swaping will be performed
+# this way the largest element of the array is place at the end of the array
+def bubble_sort(array)
+  outer_loop_guard = 1
+  (array.size - 1).times do
+    inner_loop_guard = 0
+    while inner_loop_guard < array.size - outer_loop_guard
+      if yield(array[inner_loop_guard], array[inner_loop_guard + 1])
+        array[inner_loop_guard], array[inner_loop_guard + 1] = array[inner_loop_guard + 1], array[inner_loop_guard]
+      end
+      inner_loop_guard += 1
+    end
+    outer_loop_guard += 1
+  end
+  array
+end
```
* [ ] `bubble_sort` method must be implemented and working ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sort.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-1))
```ruby
unsorted = (1..10).to_a.reverse!
bubble_sort(unsorted) != unsorted.sort
```

temesghentekeste commented:
Thanks, @suretrust for your valuable feedback.

To address your issues, we have added the bubble_sot_by method and implemented the functionality of both methods inside main.rb. We also fixed the linter errors.

kindly request our project.
